packages/block-editor/src/components/caption/index.native.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { RichText } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+const Caption = ( { accessible, accessibilityLabel, onBlur, onChange, onFocus, isParentSelected, isSelected, text } ) => {
```

Comment:
`isParentSelected`
It feels "weird" the need to know about the parent's state, but I'm not sure if it's a bad practice or not.
What do you think about passing a `hidden` prop instead (or similar)?

cc @Tug what do you think?

Reply:
I'd say it depends, usually we'd want to use an `isVisible` props in children to avoid unmounting a component from the dom or similar things, it's not a very common thing though.
In this case I think we could do without. just have `{ isSelected && <Caption ... /> }' would make the code simpler (less dependencies).

Reply:
>It feels "weird" the need to know about the parent's state, but I'm not sure if it's a bad practice or not.
>What do you think about passing a hidden prop instead (or similar)?

It does feel a bit weird, I agree. I think the weirdness (for me at least) just comes from explicitly referencing the parent in the prop name. If I understand correctly, you're suggesting we could pass down a prop specifying whether the child should be hidden? My concern with that was that I wanted to pull the logic for whether the child should be hidden ([line 14](https://github.com/WordPress/gutenberg/pull/16825/files#diff-fb4bb656dce8b784120b171a9b46cbcaR14)) into this component instead of having it separately implemented in any block that contains captions. Obviously, it's just one line, so it wouldn't be a big deal to pull it up into the (currently two) components that use captions, but I liked this solution better myself.

>In this case I think we could do without. just have `{ isSelected && <Caption ... /> }' would make the code simpler (less dependencies).

That is similar to what we had before [`{ ! RichText.isEmpty( caption ) > 0 || isSelected ) && (<Caption .../>}`](https://github.com/WordPress/gutenberg/pull/16759/files#diff-a61b3f346c11f42236eb2b1a783bf9bfL341), but I had to [change from that to using the `display` style](https://github.com/WordPress/gutenberg/pull/16759) to show/hide the caption to avoid a keyoard dismissal bug on Android.